### DISCUSSION TITLE
在 IOS 手机上运行 CloudflareST 的简单教程~

### DISCUSSION THREAD

**Discussion by kanomahoro** [XIU2/CloudflareSpeedTest#321] [分享]
~~经过测试 CFST 可以在 IOS 设备上正常使用。~~

### \# 注意！iSH Shell 新版本似乎存在一些问题，会导致 CFST 似乎无法正常运行，暂无解决办法

> 在 **Android 手机**上独立运行 CFST 的教程：#61、#320    

****

## \# 安装 iSH Shell (Linux 微型虚拟机)
在 App Store 中搜索并安装 ` iSH Shell ` App。

<img src="https://user-images.githubusercontent.com/88894099/147058962-d62831b1-a9a2-4887-8c75-21d8b3425405.jpg" width="40%">

下载安装后打开，这时候就会看到熟悉 Linux 系统！接下来就和 Linux 系统上的下载、运行步骤一样了。

****

## \# 安装 CFST

<details>
<summary><strong><code>非必需的创建文件夹步骤...</code></strong></summary>

****

毕竟手机上打字不方便，所以我就**省略了创建文件夹的步骤**，大家有强迫症的可以在安装 CFST **之前**先执行：

``` yaml
# 如果是第一次使用，则建议创建新文件夹（后续更新请跳过该步骤）
mkdir cfst

# 进入文件夹（后续更新时，只需要进入文件夹，然后重复上面那几个下载、解压命令即可）
cd cfst
```
</details>

``` yaml
# 下载 CFST 压缩包（必须下载 i386 版本，这个地址是始终指向最新版本的）
wget -N https://github.com/XIU2/CloudflareSpeedTest/releases/latest/download/cfst_linux_386.tar.gz

# 考虑到国内直接从 Github 下载速度很慢甚至连不上，那么请尝试使用下面这几个镜像加速之一：
# wget -N https://wget.la/https://github.com/XIU2/CloudflareSpeedTest/releases/latest/download/cfst_linux_386.tar.gz
# wget -N https://ghfast.top/https://github.com/XIU2/CloudflareSpeedTest/releases/latest/download/cfst_linux_386.tar.gz
# wget -N https://ghproxy.it/https://github.com/XIU2/CloudflareSpeedTest/releases/latest/download/cfst_linux_386.tar.gz
# wget -N https://gh-proxy.org/https://github.com/XIU2/CloudflareSpeedTest/releases/latest/download/cfst_linux_386.tar.gz
# wget -N https://cdn.gh-proxy.org/https://github.com/XIU2/CloudflareSpeedTest/releases/latest/download/cfst_linux_386.tar.gz

# 提示：wget 的 -N 参数是为了下载新版本时（文件名都一样），直接覆盖压缩包文件，避免被重命名加上 .1 后缀
# 如果下载失败的话，尝试删除 -N 参数（如果是为了更新，则记得提前删除旧压缩包 rm cfst_linux_386.tar.gz ）
https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.3.4/cfst_linux_386.tar.gz

# 解压
tar -zxf cfst_linux_386.tar.gz

# 赋予执行权限
chmod +x cfst
```

****

## \# 运行 CFST

``` yaml
# 运行（不带参数）
./cfst

# 运行（带参数示例）
./cfst -dd -tll 90
```
****

## \# 注意事项

- 注意 iSH Shell 这个 Linux 微型虚拟机底层是二进制转译 i686，因此只支持 linux 386 版本的 CFST。
- 所有文件都是安装在虚拟机内的，因此不需要的时候，直接卸载 ` iSH Shell ` 即可，虚拟机内的所有文件也都会被一同删除。

**Comment by XIU2** [comment]
你这排版看着好别扭，我帮你优化下排版吧~

**Reply by liuyichenchen** [reply]
在ios上运行乱码

**Comment by pingshu** [comment]
ipad上运行显示I/O error

**Comment by 007xyz** [comment]
大佬，ios还是下载不了

**Comment by hongli11** [comment]
？？

**Comment by asiv580** [comment]
下载命令应该是wget https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.0.2/CloudflareST_linux_386.tar.gz

**Comment by XIU2** [comment]
@asiv580 没有错，wget 加上 ` -N ` 参数是为了实现**下载新版本时直接覆盖**。

比如你先下载一个 2.0.1 版本的，然后再去下载一个 2.0.2 版本的，不同版本的压缩包文件名都一样。

## \# 不加 ` -N ` 参数

如果不加 ` -N ` 参数，因为文件名一样，所以 wget 会在下载 2.0.2 版本时**重命名压缩包文件**（文件名后面加上 ` .1 `）。

因此还需要提前删除旧压缩包文件，或下载后再删除旧压缩包文件并重命名新压缩包文件。。。

``` bash
root@A:~/# wget https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.0.1/CloudflareST_linux_386.tar.gz
--2022-01-10 10:59:17--
... ...
Length: 2169371 (2.1M) [application/octet-stream]
Saving to: ‘CloudflareST_linux_386.tar.gz’

CloudflareST_linux_386.tar.gz 100%[=================================================>]   2.07M  --.-KB/s    in 0.06s

2022-01-10 10:59:18 (32.7 MB/s) - ‘CloudflareST_linux_386.tar.gz’ saved [2169371/2169371]



root@A:~/# wget https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.0.2/CloudflareST_linux_386.tar.gz
--2022-01-10 10:59:24--
... ...
Length: 2170442 (2.1M) [application/octet-stream]
Saving to: ‘CloudflareST_linux_386.tar.gz.1’

CloudflareST_linux_386.tar.gz 100%[=================================================>]   2.07M  --.-KB/s    in 0.05s

2022-01-10 10:59:24 (43.7 MB/s) - ‘CloudflareST_linux_386.tar.gz.1’ saved [2170442/2170442]
```

****

## \# 加上 ` -N ` 参数

如果加上 ` -N ` 参数，wget 会对比本地文件 与 远程文件的时间，如果要下载的文件更新一些，那么即使文件名一样，wget 也会在下载 2.0.2 版本时**直接覆盖**（如果时间一样的话则不会下载，毕竟这也没意义）。

这样就避免了提前删除旧压缩包文件的步骤（或下载后再去重命名的步骤）。

``` bash
root@A:~/# wget -N https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.0.1/CloudflareST_linux_386.tar.gz
--2022-01-10 10:56:23--
... ...
Length: 2169371 (2.1M) [application/octet-stream]
Saving to: ‘CloudflareST_linux_386.tar.gz’

CloudflareST_linux_386.tar.gz 100%[=================================================>]   2.07M  --.-KB/s    in 0.07s

2022-01-10 10:56:24 (30.3 MB/s) - ‘CloudflareST_linux_386.tar.gz’ saved [2169371/2169371]



root@A:~/# wget -N https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.0.2/CloudflareST_linux_386.tar.gz
--2022-01-10 10:56:32--
... ...
Length: 2170442 (2.1M) [application/octet-stream]
Saving to: ‘CloudflareST_linux_386.tar.gz’

CloudflareST_linux_386.tar.gz 100%[=================================================>]   2.07M  --.-KB/s    in 0.06s

2022-01-10 10:56:33 (32.4 MB/s) - ‘CloudflareST_linux_386.tar.gz’ saved [2170442/2170442]
```

**Comment by XIU2** [comment]
@ventusoon 要么你禁用了下载测速（`-dd` 参数），要么你遇到了被假墙的 IP，如果你不是移动，那么请用 `-tll` 参数过滤低延迟的 IP（例如 `-tll 110` 即过滤低于 110ms 的 IP）。
移动直连香港，会遇到低延迟的香港 IP，而电信联通扫到最低延迟的 IP 普遍都是欧美的，延迟最低也不会低于 130ms。

被假墙的 IP 特征：
 - ICMP Ping 延迟正常，TCP Ping 端口延迟异常的低（因为是在出口处 TCP 劫持，所以延迟 = 你到出口处的距离）。
 如果你遇到的是被假蔷的 IP，那么你距离出口处还挺远的。。。

**Comment by ventusoon** [comment]
关了代理速度还是0

**Comment by XIU2** [comment]
@ventusoon 你重复发同一张图没有意义，看我前面说的话。

**Comment by tianchixia** [comment]
<img src="https://user-images.githubusercontent.com/71068929/150881110-7b62cc51-4c24-42cb-baea-df3ea46ce003.png" width="30%">

**Reply by LCX66** [reply]
怎么到这一步

**Reply by LCX66** [reply]
能不能麻烦说一下步骤

**Reply by cy920820** [reply]
它的版本应该是 1.2.3

**Comment by tianchixia** [comment]
苹果手机📱上测试出来的ip速度每次都不到1mb/s，同一个网络下，电脑上测出来的能到10md/s
以上，这是为什么？

**Comment by XIU2** [comment]
@tianchixia 不清楚，我没有 IOS 设备。

**Comment by tianchixia** [comment]
恩，那目前就先在电脑上优先吧，对了，有没有单独测试某一个节点的速度的方法？

**Comment by XIU2** [comment]
@tianchixia https://github.com/XIU2/CloudflareSpeedTest#-单独对一个或多个-ip-测速

**Comment by tgrocp** [comment]
大佬。mac系统怎么用啊

**Reply by wantaiyu** [reply]
![IMG_0706](https://github.com/XIU2/CloudflareSpeedTest/assets/48817344/8d317a9d-0d1c-42de-b536-2a2b576efb6e)
不懂哪个步骤出问题了

**Reply by HangJau** [reply]
这个问题是ish的问题。ish 的iuss中有这个问题，我看还没解决一直挂着的

**Reply by shylock-wu** [reply]
解决了吗，时间：`2023年11月07日10:45:25`。还没解决

**Reply by HangJau** [reply]
还没有

**Reply by wantaiyu** [reply]
下载一个旧版本的就可以了
![1659](https://github.com/XIU2/CloudflareSpeedTest/assets/48817344/54df4e79-57f5-4b77-8fa6-4eb198fc2544)

**Reply by HangJau** [reply]
老哥，怎么下载的旧版本？是通过助手吗？

**Reply by wantaiyu** [reply]
你去吾爱破解论坛搜索这个：iOS任意版本号APP下载v5.3。有教程，很简单。

**Reply by 550530** [reply]
> 你去吾爱破解论坛搜索这个：iOS版本号APP下载v5.3。有教程，很简单。
你好 请问ish需要哪个版本啊 我下载了好几个版本还是提示和这个差不多的错误 要不就是卡在0%不动

**Comment by XIU2** [comment]
@tgrocp 和 Linux 差不多啊，下载、解压、进入目录、赋予执行权限，然后 ./CloudflareST 执行就行了，经常用 MAC 的不应该不知道吧。。。
具体的话我也不清楚，我没有 MAC 设备。

**Comment by tgrocp** [comment]
真的不知道。半桶水啊我，能出个教程吗大神？

**Comment by LYOyoo** [comment]
为啥我下载了解压的时候提示no such file or directory

**Comment by XIU2** [comment]
@LYOyoo 这个报错是找不到压缩包文件，因此有两种可能：
1. 你**文件下载失败**了，所以提示找不到压缩包文件。
2. 你文件下载成功了，但是你的解压命令中的**文件名写错**了，所以提示找不到压缩包文件。。

**Comment by LYOyoo** [comment]
下载失败了，-n去掉就好了

**Comment by 404user** [comment]
学习了，感谢啊。这样在手机上操作就方便多了。

**Comment by z10056** [comment]
> 苹果手机📱上测试出来的ip速度每次都不到1mb/s，同一个网络下，电脑上测出来的能到10md/s
> 以上，这是为什么？

我和你一样 电脑五百兆宽带 小米10和电脑能跑到四五十MB 在这上面只能跑到1MB

**Comment by XIU2** [comment]
@z10056 电脑和安卓都挺快，就 IOS 这个虚拟机下速度不行？安卓用的是 Termux 虚拟机么？
可以找找 IOS 应用商店里还有没有其他类似虚拟机 APP。。。

**Comment by z10056** [comment]
> Termux

安卓用的是Termux  iOS上除了ish之外找不到其他的Linux虚拟机了 付费的不知道 还有一个utm虚拟机 只不过得用电脑来安装 目前手头上没电脑  哎

**Comment by 007xyz** [comment]
> 不要默认测10个，设置一下测50个，我测50个选出几个50mb的

**Comment by z10056** [comment]
> 不要默认测10个，设置一下测50个，我测50个选出几个50mb的

宽带还行 流量顶不住 还是搞个安卓备用机跑这个吧

**Comment by ElonWang8** [comment]
mkdir: can't create directory 'CloudflareST': File exists

**Comment by ElonWang8** [comment]
> mkdir：无法创建目录'CloudflareST'：文件存在

我这里ios显示无法创建是因为啥呀大神

**Comment by XIU2** [comment]
@ElonWang8 你提问前就不能自己翻译一下吗？。。。这就是提示你**文件夹已存在**，无法重复新建文件夹了呀。这个步骤不是必须的，提供给有强迫症人用的。

**Comment by slarkleo** [comment]
IOS 怎么修改hosts呢 ？

**Comment by fenghengwei** [comment]
ipad上运行显示I/O error

**Comment by TheKing1991** [comment]
苹果安装wget是apk add wget

**Reply by v0re** [reply]
curl -O -L url

**Comment by sunshowerstack** [comment]
iOS上测出来最快的ip是给电脑用还是手机用。给电脑用的话作用不大，电脑上直接运行就可以了。如果给流量下的手机用，那怎么写到手机上？在ish里写入/etc/hosts？感觉不靠谱啊

**Comment by Sunbangyan233** [comment]
> IOS 怎么修改hosts呢 ？

去App store下个<b>Stream</b>，首页就有hosts设置，添加后开抓包安装配置信任证书即可。
例如这样：![D86A117F-21B2-4FA2-9D14-172A1B9B5092](https://pics.sunbangyan.cn/i/2022/09/03/3vdss.jpeg)
如果某些网站或App无法正常使用，尝试在抓包设置中开启白名单模式（需要再手动添加一遍自选IP域名）试试。
~~不过是不是偏离主题了~~

**Comment by LeonLeeLi** [comment]
它提示权限不够
Permission denied

**Comment by LeonLeeLi** [comment]
[![zkcdfO.png](https://s1.ax1x.com/2022/11/14/zkcdfO.png)](https://imgse.com/i/zkcdfO)

**Comment by XIU2** [comment]
@LeonLeeLi 
你是不是漏了 赋予执行权限 的步骤？
```
chmod +x CloudflareST
```

**Comment by XIU2** [comment]
@wozhishilaonanhai 当然可以，不过这就和本 Issues 目的完全不一样了。

因为 `电脑连接手机热点` 来测速，本质上还是**在电脑上测速**，不过走的是移动网络。
而本 Issues 重点/目标是为了**在 IOS 手机上独立测速**，并不关注是走的什么网络。

**Comment by lanling175** [comment]
Ios15.4.1 ./CloudflareST 命令一执行就闪退

**Comment by znjuery** [comment]
[按照2.2.3后，执行./CloudflareST，显示报错如下：[](url)](url)
![1](https://user-images.githubusercontent.com/32011913/234471203-74153352-0dbf-4f6d-a093-8ff30e892922.jpg)

**Reply by XIU2** [reply]
这种问题比较罕见，你先试试旧版本是否可以运行。

**Reply by jarvie-mei** [reply]
我也出现了同样的问题

**Reply by coco-coc** [reply]
这个是ish更新后出现这样，没更新前是正常的

**Reply by Don7578** [reply]
> 这个是ish更新后出现这样，没更新前是正常的

中间iSH更新了一次，问题依然没有修复，又下不了旧版，一直用不了……

**Reply by wantaiyu** [reply]
我也是用着用着就突然不行了，卸载后重新下载也出现你这种情况，安卓的也是，以前好好的，现在测出来下载丢包没有一个能用的。只有电脑版没问题。

**Reply by bjmbdy** [reply]
2.2.4同样是这个错误，用旧版本也不行，好像确实是iSH更新后的问题。

**Reply by jackoma1989** [reply]
一样

**Reply by RSJYPJY** [reply]
有没有解决问题

**Reply by wantaiyu** [reply]
下旧版本
![1659](https://github.com/XIU2/CloudflareSpeedTest/assets/48817344/c23e3fdb-0421-4e93-8d7a-3b23d263356e)

**Comment by wenwei0486** [comment]
更新以后测速 闪退

**Reply by jzsmqss** [reply]
报错信息太多了，好几页。 期待什么时候修复

**Reply by Don7578** [reply]
还是研究怎么恢复旧版本iSH好过（通过旧版iTunes）

**Reply by RSJYPJY** [reply]
> 还是研究怎么恢复旧版本iSH好过（通过旧版iTunes）

不行，今天我捣鼓一天了，旧版itunes安装不上了

**Reply by coco-coc** [reply]
依然可以

**Reply by Don7578** [reply]
给后来人看：吾爱破解站内搜索time brand，按教程下载旧版iSH 1.22，正在正常使用中。有别的办法，例如以前有备份过旧版本，也可以。总之要降级。

**Comment by tosasitill** [comment]
现在ish执行到500个ip延迟测试左右就卡住了 貌似是什么限制 重启也不行 最多500个

**Comment by coco-coc** [comment]
ish旧版好点，新版会有问题

**Comment by coco-coc** [comment]
建议回到1.2.3版本

**Reply by tosasitill** [reply]
![IMG_5936](https://github.com/XIU2/CloudflareSpeedTest/assets/66861785/cf239e83-8e2e-4ff1-bf56-ed901dd5bc08)
难受 有什么办法吗

**Reply by coco-coc** [reply]
只能降级

**Reply by weige112233** [reply]
为什么没有下载速度

**Comment by cy920820** [comment]
确实 ish bug 很多，或者说还是需要调试
我现在直接让 iPhone 热点给 mac，在 mac 上测速完之后直接给到 手机端

**Reply by Don7578** [reply]
iOS不方便更换ip库。请问在Mac上怎么测？跟linux一样吗？

**Reply by XIU2** [reply]
@Don7578 MAC 下操作命令行程序（也就是 CloudflareST 这样的）和 Linux 没区别，命令什么的几乎一模一样。

**Comment by wenzai007** [comment]
想问下有必要这么麻烦么？？   如果我手机开一个热点， 电脑连上这个热点， 在电脑上测试，是不是就行了？

**Reply by cy920820** [reply]
自信点

**Reply by mraxu** [reply]
完全可以

**Comment by 3117310** [comment]
请问NewTerm是什么微型机，不用386可以吗？测试加端口可以吗？

**Comment by git1568** [comment]
ios怎么更换ip？

**Comment by cyp-vet** [comment]
第一步：下载
wget https://github.com/XIU2/CloudflareSpeedTest/releases/download/v2.3.4/cfst_linux_386.tar.gz

第二步：解压：
tar -zxf cfst_linux_386.tar.gz

第三步：赋予执行权限：
chmod +x cfst

第四步：运行程序：
./cfst

Or：带参数运行：
./cfst -dd -tll 90
